enterprise/frontend/src/metabase-enterprise/documents/components/Editor/hooks/useCommentsButton.tsx

```diff
@@ -0,0 +1,39 @@
+import { t } from "ttag";
+
+import Link from "metabase/common/components/Link";
+import { Icon } from "metabase/ui";
+
+interface CommentsButtonOptions {
+  active: boolean;
+  disabled: boolean;
+  href: string;
+  unresolvedCommentsCount: number;
+}
+
+export const useCommentsButton = ({
+  active,
+  disabled,
+  href,
+  unresolvedCommentsCount,
+}: CommentsButtonOptions) => {
+  const hasUnresolvedComments = unresolvedCommentsCount > 0;
+  return {
+    ...(!disabled
+      ? {
+          component: Link,
+          // If no existing unresolved comments comments, add query param to auto-open new comment form
+          to: hasUnresolvedComments ? href : `${href}?new=true`,
+        }
+      : {}),
+    disabled,
+    leftSection: (
+      <Icon name={hasUnresolvedComments ? "comment" : "add_comment"} />
+    ),
+    px: "sm",
+    size: "xs",
+    bd: 0,
+    "aria-label": t`Comments`,
+    variant: active ? "filled" : "default",
+    children: hasUnresolvedComments ? unresolvedCommentsCount : null,
+  };
+};
```

Comment:
This should be a react component.

Reply:
> We still have to use specific props, rather than a component from this hook.

Only this part:
```tsx
hasUnresolvedComments ? href : `${href}?new=true`
```

>  from this hook

This is not a hook.

----

Here: abd2c740212303e2b0a7eb883ed00f966a5bf39